diesel_cli/tests/migration_generate.rs

```diff
@@ -0,0 +1,29 @@
+use support::{TestEnvironment};
+
+use std::process::Command;
+use std::path::PathBuf;
+use std::{env, fs};
+
+#[test]
+fn diesel_migration_generate_makes_valid_migration_directory() {
+    let test_environment = TestEnvironment::new();
+
+    let migrations_dir = test_environment.root_path().join("migrations");
+    fs::create_dir(&migrations_dir).unwrap();
+
+    let diesel_exe = env::current_exe().unwrap().parent().unwrap().join("diesel");
+    let mut command = Command::new(diesel_exe);
+    command.arg("migration")
+        .arg("generate")
+        .arg("create_posts_table")
+        .current_dir(&test_environment.root_path());
+    command.output().unwrap();
+
+    let create_posts_dir = fs::read_dir(&migrations_dir).unwrap().nth(0).unwrap().unwrap().path();
+    let filenames: Vec<String> = create_posts_dir.read_dir()
```

Comment:
The style of a lot of these tests bugged me. I just pushed some stuff up to master to demonstrate what I'd like the style of these tests to look like. Mainly I want the signal to noise ratio to be a bit better.